src/aiperf/gpu_telemetry/amdsmi_collector.py

```
@@ -0,0 +1,380 @@
+
+    async def is_url_reachable(self) -> bool:
+        """Check if AMDSMI is available and at least one GPU is visible.
+
+        Returns:
+            True if amdsmi can initialize and enumerates >=1 processor handle.
+        """
+        if self._initialized:
+            return len(self._gpus) > 0
+        try:
+            return await asyncio.to_thread(self._probe_devices)
+        except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)